xgi/generators/uniform.py

```diff
@@ -89,3 +100,253 @@ def uniform_hypergraph_configuration_model(k, m, seed=None):
+    H.add_nodes_from(node_labels)
+
+    block_range = range(len(sizes))
+    # Split node labels in a partition (list of sets).
+    size_cumsum = [sum(sizes[0:x]) for x in range(0, len(sizes) + 1)]
+    partition = [
+        list(node_labels[size_cumsum[x] : size_cumsum[x + 1]])
+        for x in range(0, len(size_cumsum) - 1)
+    ]
+
+    for block in itertools.product(block_range, repeat=m):
+        if p[block] == 1:  # Test edges cases p_ij = 0 or 1
+            edges = itertools.product((partition[i] for i in block_range))
+            for e in edges:
+                H.add_edge(e)
+        elif p[block] > 0:
+            partition_sizes = [len(partition[i]) for i in block]
+            max_index = reduce(operator.mul, partition_sizes, 1)
+            if max_index < 0:
+                raise Exception("Index overflow error!")
+            index = np.random.geometric(p[block]) - 1
+
+            while index < max_index:
+                indices = _index_to_edge_partition(index, partition_sizes, m)
+                e = [partition[block[i]][indices[i]] for i in range(m)]
+                if len(e) == m:
+                    H.add_edge(e)
+                index += np.random.geometric(p[block])
+    return H
+
+
+def uniform_hypergraph_planted_partition_model(n, m, rho, k, epsilon):
+    """Construct the m-uniform hypergraph planted partition model (m-HPPM)
+
+    Parameters
+    ----------
+    n : int > 0
+        Number of nodes
+    m : int > 0
+        Hyperedge size
+    rho : float between 0 and 1
+        The fraction of nodes in community 1
+    k : float > 0
+        Mean degree
+    epsilon : float > 0
+        Imbalance parameter
+
+    Returns
+    -------
+    Hypergraph
+        The constructed m-HPPM hypergraph.
+
+    Raises
+    ------
+    XGIError
+        - If rho is not between 0 and 1
+        - If the mean degree is negative.
+        - If epsilon is not between 0 and 1
+    
+    See Also
+    --------
+    uniform_SBM_hypergraph
+
+    References
+    ----------
+    Nicholas W. Landry and Juan G. Restrepo.
+    "Polarization in hypergraphs with community structure."
+    Preprint, 2023. https://doi.org/10.48550/arXiv.2302.13967
+    """
+
+    if rho < 0 or rho > 1:
+        raise XGIError("The value of rho must be between 0 and 1")
+    if k < 0:
+        raise XGIError("The mean degree must be non-negative")
+    if epsilon < 0 or epsilon > 1:
+        raise XGIError("epsilon must be between 0 and 1")
+
+    sizes = [int(rho * n), n - int(rho * n)]
+
+    p = k / (m * n ** (m - 1))
+    # ratio of inter- to intra-community edges
+    q = rho**m + (1 - rho) ** m
+    r = 1 / q - 1
+    p_in = (1 + r * epsilon) * p
+    p_out = (1 - epsilon) * p
+
+    p = p_out * np.ones([2] * m)
+    p[tuple([0] * m)] = p_in
+    p[tuple([1] * m)] = p_in
+
+    return uniform_SBM_hypergraph(n, m, p, sizes)
+
+
+def uniform_erdos_renyi_hypergraph(n, m, k):
+    """Generate an m-uniform Erdős–Rényi hypergraph
```

Comment:
This is different from our non-uniform version of an ER hypergraph which takes `p` rather than `k` as an argument. Do we want to make it consistent? Or offer both options?

```suggestion
    """Generate an m-uniform Erdős–Rényi hypergraph.
    
    This creates an hypergraph with `n` nodes where hyperedges of size `m` created at random to obtain a mean degree of `k`. 
```